### PR TITLE
Support for Trezor

### PR DESCRIPTION
**Pull request by StevenJNPearce**
### PR Checklist (check all)

* [Y] Updated `CHANGELOG.md` to describe the included fixes and changes made
* [Y] Included issue number on the description of the PR
* [Y] Commented on the relevant issue thread about this PR

### Issue number

#182 

### Changelog

Support for Trezor

### CODE REVIEW

**Comment by ricburton** [comment]
🙌 Preview this pull request.

Built with commit e866e5cbc42367243fc31ec634d7960d905c71cb

https://deploy-preview-270--balance-manager.netlify.com

**Comment by ricburton** [comment]
@jinchung I am going to do a full review of this with a model T and normal Trezor. Will take a little while.

**Comment by ricburton** [comment]
There's a typo here:

Please connect your *Tezor* and follow the instructions

Also, the Model T does not connect. It works on MyCrypto. About to try the original Trezor.

**Comment by ricburton** [comment]
Trezor One fails too. 

Anything I can do to help here?

screen shot 2018-06-09 at 20 06 29

**Comment by ricburton** [comment]
I update the support article: https://support.balance.io/manager/how-do-i-connect-my-trezor

**Comment by StevenJNPearce** [comment]
I think it's not working because the CI sever hasn't been whitelisted by trezor, running locally there are no problems.

**Comment by ricburton** [comment]
Ah got it! Excitedly checking this out now @StevenJNPearce.

**Comment by ashchan** [comment]
[hdhey](https://github.com/cryptocoinjs/hdkey) is a standalone NPM package, I think this can use that instead of including `ethereumjs-wallet`?

File: package.json [hunk 1]
```diff
@@ -18,6 +18,8 @@
     "axios": "^0.18.0",
     "bignumber.js": "^7.0.1",
     "ethereumjs-tx": "^1.3.4",
+    "ethereumjs-wallet": "^0.6.0",
```
Comment (jinchung) [inline package.json:21]: Could we remove this dependency? It looks like you're already pulling in hdkey as a dependency, and it looks like that's the only reason you have the ethereumjs-wallet.

File: src/languages/_english.json [hunk 2]
```diff
@@ -109,9 +115,11 @@
       "click_to_copy_to_clipboard": "Click to copy to clipboard",
       "failed_request": "Failed request, please refresh",
       "failed_ledger_connection": "Failed to connect to Ledger, please check your device",
+      "failed_trezor_connection": "Failed to connect to Trezor, please check your device",
       "no_transactions": "No transactions found for this account",
       "no_interactions": "No interactions found for this account",
-      "please_connect_ledger": "Please connect and unlock Ledger then select Ethereum"
+      "please_connect_ledger": "Please connect and unlock Ledger then select Ethereum",
+      "please_connect_trezor": "Please connect your Tezor and follow the instructions"
```
Comment (jinchung) [inline src/languages/_english.json]: Tezor --> Trezor

File: src/reducers/_trezor.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      dispatch({ type: TREZOR_CONNECT_FAILURE });
+    });
+};
+
+export const trezorUpdateNetwork = network => (dispatch, getState) => {
+  const accountAddress = getState().account.accountAddress;
+  dispatch({ type: TREZOR_UPDATE_NETWORK, payload: network });
+  dispatch(accountUpdateNetwork(network));
+  dispatch(accountUpdateAccountAddress(accountAddress, 'TREZOR'));
+  dispatch(trezorConnectInit())
```
Comment (jinchung) [inline src/reducers/_trezor.js]: Nice - I think we're missing this in our Ledger logic for "update network" 😅

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 
 ### Added
 
-*
+* Support for Trezor
```
Comment (jinchung) [inline CHANGELOG.md]: @StevenJNPearce thanks a lot for your work!

**Review by jinchung — APPROVED** [review]

**Pull request merged**